File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // assign key of the cipher to integer variable 'key'
    int key = atoi(argv[1]);

    // check if two command line arguments have been entered
    if (argc == 2)
    {
        // get a string from user and print the ciphered string with code that follows
        string code = get_string("plaintext: ");
        printf("ciphertext: ");

        int n = strlen(code);
        int cipher[n];

        for (int i = 0; i < n; i++)
        {
            // check if character in the plaintext is a lowercase letter
            if (islower(code[i]))
            {
                // update cipher array with transformed letter and print it
                cipher[i] = ((code[i] + key - 'a') % 26) + 97;
                printf("%c", cipher[i]);
            }
            // check if character in the plaintext is an uppercase letter
            else if (isupper(code[i]))
            {
                // update cipher array with transformed letter and print it
                cipher[i] = ((code[i] + key - 'A') % 26) + 65;
                printf("%c", cipher[i]);
            }
            // if the character is neither a lowercase or uppercase letter, print it as-is
            else
            {
                cipher[i] = code[i];
                printf("%c", cipher[i]);
            }
        }
        printf("\n");
        return 0;
    }
    // if two command line args aren't enter, print error message
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}

File: pset6/caesar.py
from cs50 import get_string
from sys import argv

def main():

    if len(argv) != 2:
        sys.exit("Usage: python caesar.py key")


    key = int(argv[1])
    code = get_string("plaintext: ")
    print("ciphertext: ", end="")

    for c in code:

        if c.islower():

            c_cipher = ((ord(c) + key - ord('a')) % 26) + 97
            print(f"{chr(c_cipher)}", end="")

        elif c.isupper():

            c_cipher = ((ord(c) + key - ord('A')) % 26) + 65
            print(f"{chr(c_cipher)}", end="")

        else:

            if (ord(c) >= 65 and ord(c) <= 90) or (ord(c) >= 97 and ord(c) <= 122):
                print(f"{chr(c)}", end="")
            else:
                print(c, end="")

    print()
    return 0

if __name__ == "__main__":
    main()